datadog_checks_base/datadog_checks/base/checks/openmetrics/mixins.py

```diff
@@ -13,8 +13,7 @@
 from six import PY3, iteritems, string_types
 
 from .. import AgentCheck
-
-from datadog_checks.config import is_affirmative
+from ...config import is_affirmative
```

Comment:
what are the benefits of using relative imports in general?

Asking mostly because I find them harder to read. For example here:

```
from ....config import is_affirmative
```

would reintroduce the bug. The only difference is one dot `.` is the relative path.

Reply:
Echoes my feelings 100% - we are putting in a patch to fix something that seems to me be may be more of a fundamental issue that might require a more elaborate fix.

Reply:
Two benefits I see immediately: 
 - no name conflict between packages and file names
 - it's clear at a glance you're importing something from the same package

For more details, I recommend reading
https://www.python.org/dev/peps/pep-0328/

The one dot difference would not really be the same bug. This import would always fail regardless of the import order

Reply:
thanks for the context 👍

I'd still argue that it's the same bug though: `datadog_checks.config` and `datadog_checks.base.config` both exist, and are part of the same top-level package (`datadog_checks`). So, from `datadog_checks.base`, depending on whether you use an additional dot `.` or not in your relative import, you'll either import `datadog_checks.config` or `datadog_checks.base.config`.

In general, fundamentally, using relative imports from within the `datadog_checks.base` package doesn't prevent python code from `datadog_checks.base` to import paths in `datadog_checks` that are not under `base`, so they don't prevent circular dependencies like the present one.

I'm not arguing that relative imports should be banned/avoided though :) I mostly want to point out that the pattern will not prevent such circular import issues from happening in the future. If you have thoughts/ideas on how these could be caught by a linter/test, I'd be glad to hear them!

Reply:
Oh indeed you're right, I got confused.
Yeah we are in this situation because we originally didn't have the `base` sub package, and when we had to introduced it, we also had to keep the old structure and introduce redirection in order not to break any custom checks. I guess the only real long term solution would be to finally get rid of the old structure in a major version.
In the meantime, yeah, we probably want to find a way to test for that.

Reply:
I need Aspirine now!

Reply:
💊